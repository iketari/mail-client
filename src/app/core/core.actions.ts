import { Action } from '@ngrx/store';

export enum CoreActionTypes {
  LoadCores = '[Core] Load Cores'
}

export class Core implements Action {
  readonly type = CoreActionTypes.LoadCores;
}

export type CoreActions = LoadCores;
