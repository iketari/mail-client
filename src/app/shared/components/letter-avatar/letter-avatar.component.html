<span class="email-avatar"
  [ngStyle]="{'background-color': stringToColour(name)}">{{getInitials()}}</span>