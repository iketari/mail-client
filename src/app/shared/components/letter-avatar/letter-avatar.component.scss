.email-avatar {
  display: inline-block;
  width: 3.125em;
  height: 3.125em;
  line-height: 3.125em;
  margin-right: 0.5em;
  text-align: center;
  text-transform: uppercase;
  border-radius: 100%;
  background-color: #28558f;
  color: #cecece;
}