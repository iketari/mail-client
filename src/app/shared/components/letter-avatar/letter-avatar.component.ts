import {
  Component,
  OnInit,
  ChangeDetectionStrategy,
  Input,
  ViewEncapsulation
} from '@angular/core';

const colours = [
  [['#5e0231'], ['#dbc3d0']],
  [['#283018'], ['#f6f1ed']],
  [['#aa863a'], ['#f6f1ed']],
  [['#729f98'], ['#f6f1ed']],
  [['#1d2120'], ['#f6f1ed']],
  [['#5a5c51'], ['#f6f1ed']],
  [['#ba9077'], ['#f6f1ed']],
  [['#e6cf8b'], ['#e8edf3']],
  [['#b56969'], ['#e8edf3']],
  [['#22264b'], ['#e8edf3']],

  [['#edd9c0'], ['#f6f1ed']],
  [['#c9d8c5'], ['#f6f1ed']],
  [['#a8b6bf'], ['#f6f1ed']],
  [['#7d4627'], ['#f6f1ed']],
  [['#b3c2bf'], ['#c0dfd9']],
  [['#3b3a36'], ['#c0dfd9']],
  [['#262216'], ['#f6f1ed']],
  [['#49412c'], ['#f6f1ed']],
  [['#97743a'], ['#f6f1ed']],

  [['#b0a18e'], ['#f6f1ed']],
  [['#252839'], ['#f6f1ed']],
  [['#677077'], ['#f6f1ed']],
  [['#b5b5b7'], ['#f6f1ed']],
  [['#f2b632'], ['#f6f1ed']],
  [['#16174f'], ['#f6f1ed']],
  [['#963019'], ['#f6f1ed']],
  [['#667467'], ['#f6f1ed']],
  [['#A5DD99'], ['#f6f1ed']],
  [['#DB9B6D'], ['#f6f1ed']]
];

@Component({
  selector: 'letter-avatar',
  templateUrl: './letter-avatar.component.html',
  styleUrls: ['./letter-avatar.component.scss'],
  changeDetection: ChangeDetectionStrategy.OnPush,
  encapsulation: ViewEncapsulation.None
})
export class LetterAvatarComponent {
  @Input()
  name: string = '';

  /**
   * getInitials
   */
  public getInitials(): string {
    const parts = this.name.split(' ');
    let result = '';

    if (parts[0] && parts[1]) {
      result = parts[0].charAt(0) + parts[1].charAt(0);
    } else {
      result = this.name.slice(0, 2);
    }

    return result;
  }

  public stringToColour(str: string): string[][] {
    let hash = 0;
    for (let i = 0; i < str.length; i++) {
      hash = Math.abs(str.charCodeAt(i) + ((hash << 5) - hash));
    }

    return colours[hash % colours.length];
  }
}
