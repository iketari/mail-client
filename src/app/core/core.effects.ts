import { Injectable } from '@angular/core';
import { Actions, Effect } from '@ngrx/effects';
import { CoreActions, CoreActionTypes } from './core.actions';

@Injectable()
export class CoreEffects {

  @Effect()
  effect$ = this.actions$.ofType(CoreActionTypes.LoadCores);

  constructor(private actions$: Actions) {}
}
