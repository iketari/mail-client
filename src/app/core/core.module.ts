import { NgModule } from '@angular/core';
import { BrowserModule } from '@angular/platform-browser';
import { RouterModule } from '@angular/router';
import { StoreModule } from '@ngrx/store';

import { reducers } from './ngrx/reducers';
import { EffectsModule } from '@ngrx/effects';
import { CoreEffects } from './ngrx/effects/core.effects';
import { SharedModule } from '../shared/shared.module';
import { EmailService } from './services/email.service';
import { BackendService } from './services/backend.service';
import { SidebarComponent } from './containers/sidebar/sidebar.component';
import { NgSelectModule } from '@ng-select/ng-select';
import { FormsModule } from '../../../node_modules/@angular/forms';

@NgModule({
  declarations: [SidebarComponent],
  exports: [SidebarComponent],
  imports: [
    BrowserModule,
    StoreModule.forFeature('core', reducers),
    EffectsModule.forFeature([CoreEffects]),
    RouterModule,

    FormsModule,
    NgSelectModule,

    SharedModule
  ],
  providers: [EmailService, BackendService]
})
export class CoreModule {}
