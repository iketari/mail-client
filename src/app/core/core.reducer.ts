import { Action } from '@ngrx/store';
import { CoreActions, CoreActionTypes } from './core.actions';

export interface State {

}

export const initialState: State = {

};

export function reducer(state = initialState, action: CoreActions): State {
  switch (action.type) {

    case CoreActionTypes.LoadCores:
      return state;


    default:
      return state;
  }
}
