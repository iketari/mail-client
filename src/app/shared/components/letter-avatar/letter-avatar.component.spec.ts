import { ComponentFixture, TestBed } from '@angular/core/testing';
import { NO_ERRORS_SCHEMA } from '@angular/core';
import { LetterAvatarComponent } from './letter-avatar.component';

const pretty = require('pretty');

describe('LetterAvatarComponent', () => {
  let comp: LetterAvatarComponent;
  let fixture: ComponentFixture<LetterAvatarComponent>;

  beforeEach(() => {
    TestBed.configureTestingModule({
      declarations: [LetterAvatarComponent],
      schemas: [NO_ERRORS_SCHEMA]
    });

    fixture = TestBed.createComponent(LetterAvatarComponent);
    comp = fixture.componentInstance;
  });

  it('can load instance', () => {
    expect(comp).toBeTruthy();
  });

  it('should match snapshop', () => {
    comp.name = 'iketari@gmail.com';

    fixture.changeDetectorRef.markForCheck();
    fixture.detectChanges();
    expect(pretty(fixture.nativeElement.outerHTML)).toMatchSnapshot();
  });

  it('should match snapshop', () => {
    comp.name = 'Artsiom Mezin';

    fixture.changeDetectorRef.markForCheck();
    fixture.detectChanges();
    expect(pretty(fixture.nativeElement.outerHTML)).toMatchSnapshot();
  });
});
