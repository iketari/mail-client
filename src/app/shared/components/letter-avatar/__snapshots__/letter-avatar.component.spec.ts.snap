// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LetterAvatarComponent should match snapshop 1`] = `"<div id=\\"root1\\" ng-version=\\"6.1.0\\"><span class=\\"email-avatar\\" style=\\"color: rgb(246, 241, 237); background-color: rgb(176, 161, 142);\\" ng-reflect-ng-style=\\"[object Object]\\">ik</span></div>"`;

exports[`LetterAvatarComponent should match snapshop 2`] = `"<div id=\\"root2\\" ng-version=\\"6.1.0\\"><span class=\\"email-avatar\\" style=\\"color: rgb(246, 241, 237); background-color: rgb(181, 181, 183);\\" ng-reflect-ng-style=\\"[object Object]\\">AM</span></div>"`;
